app/services/optout/OptOutService.scala

```diff
@@ -120,6 +97,55 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     OptOutProposition(previousYearOptOut, currentTaxYearOptOut, nextTaxYearOptOut)
   }
 
+  private def makeOptOutUpdateRequestForOneYear(optOutProposition: OptOutProposition)(implicit
+                                                                                      user: MtdItUser[_],
+                                                                                      shc: HeaderCarrier,
+                                                                                      ec: ExecutionContext): Future[OptOutUpdateResponse] = {
+    val intent = optOutProposition.availableOptOutYears.head
+    makeOptOutUpdateRequest(optOutProposition, intent)
+  }
+
+  def nextUpdatesPageOneYearOptOutViewModel()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[OptOutOneYearViewModel]] = {
+    optOutOneYearViewModel((optOutData, optOutYear) => {
+      val showWarning = optOutData match {
+        case OptOutProposition(previousTaxYear, currentTaxYear, _) if previousTaxYear == optOutYear && currentTaxYear.status == Mandated => true
```

Comment:
lines 111 to 113 should be moved to inside OptOutProposition

Reply:
done